pypeit/core/wavecal/autoid.py

```diff
@@ -2480,13 +2485,13 @@ def solve_slit(self, slit, psols, msols, tcent_ecent, nstore=1, nselw=3, nseld=3
                 # First time a fit is found
                 patt_dict, final_dict = tpatt_dict, tfinal_dict
                 continue
-            elif tfinal_dict['rms'] < self._rms_threshold:
+            elif tfinal_dict['rms'] < wvutils.parse_param(self._par, 'rms_threshold', slit):
                 # Has a better fit been identified (i.e. more lines ID)?
                 if len(tfinal_dict['pixel_fit']) > len(final_dict['pixel_fit']):
                     patt_dict, final_dict = copy.deepcopy(tpatt_dict), copy.deepcopy(tfinal_dict)
         return patt_dict, final_dict
 
-    def solve_patterns(self, bestlist, tcent_ecent):
+    def solve_patterns(self, slit, bestlist, tcent_ecent):
```

Comment:
docstring?

Reply:
@rcooke-ast you wrote this routine;) I'm just fixing a bug.

Reply:
Here you go!!

        """
        Solve the patterns and find the best ID of each detected line.

        Parameters
        ----------
        slit: int
            The ID of the slit
        bestlist: list of ndarray
            A 5 element list, each containing a numpy.ndarray, with the following values required for each index:
              0: central wavelength of the pattern
              1: central dispersion of pattern
              2: sign of the pattern (note, sign = 1 [-1] if pixels correlate [anticorrelate] with wavelength
              3: index of the full list of patterns that were created from the detected arc lines
              4: index of the full list of patterns that were created from the line list.
        tcent_ecent : list
            A list [tcent, ecent] indicating which detection list
            should be used. Note that if arr_err is set then the weak
            keyword is ignored.

        Returns
        -------
        patt_dict : dict
            Dictionary containing information about the best patterns.
        """

Reply:
Added. Thanks!